internal/mining/bgblktmplgenerator_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package mining
+
+import (
+	"testing"
+	"time"
+)
+
+// TestWaitGroup ensures the API for the local waitGroup implementation behaves
+// correctly.
+func TestWaitGroup(t *testing.T) {
+
+	// goWait is a helper that calls wg.Wait() on a gorutine and closes the
+	// returned chan once Wait() returns.
+	goWait := func(wg *waitGroup) chan struct{} {
+		c := make(chan struct{})
+		go func() {
+			wg.Wait()
+			close(c)
+		}()
+
+		// Give a chance for the goroutine to execute before returning
+		// the channel.
+		time.Sleep(10 * time.Millisecond)
```

Comment:
I considered this, but also thought there's no guarantee `wg.Wait()` is actually called before `<-started` is received and `c` is returned. 

I do grant you that it's _very_ likely that will be the case though, so I'll update it.

Reply:
Right.  I agree that it's not 100% guaranteed, but neither is waiting for 10ms since it can get preempted.  This suggestion is really just based on what I've seen with CI historically where the sleeps tend to randomly cause  false positives while the channel approach has always worked in practice.